app/views/puppetclasses/_class_parameters.html.erb

```diff
@@ -1,6 +1,15 @@
 <% lookup_keys = overridable_lookup_keys(puppetclass, obj) %>
 <% lookup_keys.each_with_index do |lookup_key, index| %>
   <% lookup_value = lookup_value(obj, lookup_key) %>
+    <%
+       # LookupValue#match is generated automatically by Host#lookup_values_attributes=
+       # from the name, so don't present errors directly to the user.  Any validation
+       # errors should also be visible against the Host#name field.
+       if lookup_value.errors[:match].any?
```

Comment:
@domcleal this is what was missing and caused the test I removed to fail.
I also added back the test.